infra/Pos/Network/Windows/DnsDomains.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
+
+#if defined(mingw32_HOST_OS)
```

Comment:
Got it.

Regarding testing, I jumped over to my Windows machine and made sure it worked (prints out `10.8.4.1` IIRC, which is the same thing I get when going over to the control panel).

I tried leaving a small `makefile` along with a separate file with `main` in it so it'd be easy to test whenever someone felt like checking the C code is OK, but I couldn't make it work because
 of some header issues on my environment. As such I just left `main` commented out in the file. 

Not the best practice, I know.